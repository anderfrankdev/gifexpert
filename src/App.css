:root {
  /****** Fluid reset *******/
  /* @link https://utopia.fyi/type/calculator?c=320,15,1.2,1024,17,1.25,4,0,&s=0.75|0.5|0.25,1.5|2|3|4|6,s-l */

  --step-0: clamp(1.50rem, calc(1.05rem + 2.27vw), 2.50rem);
}

#root {
  height: 100vh;
  width: 100%;
  padding: 0rem;
  padding-block: 0;
  overflow: hidden;
  display: grid;
  grid-template-columns: 1fr;
  grid-template-rows: 
    min-content 
    min-content 
    min-content 
    min-content 
    auto;
}

.title{
  font-size: var(--step-0);
  margin-inline: 1rem;
}

.input {
  height: 2rem;
  margin-bottom: 1rem;
  font-size: 1.2rem;
  width: calc( 100% - 2rem );
  display: block;
  margin-inline: 1rem;
}

.button {
  background-color: mediumpurple;
  color: white;
  border-width: 3px;
  width: calc( 100% - 2rem );
  display: block;
  margin-inline: 1rem;
}
.button:hover {
  border-color: lightcyan;
}

.gifs-container {
  display: flex;
  justify-content: center;
  align-items: flex-start;
  flex-direction: row;
  flex-wrap: wrap;  
  width: 100%;
  overflow-y: auto;
  margin-top: 1rem;
}

.gifs-container > div {
  margin: 0.5rem;
  margin-bottom: 1rem;
}

.load-msg{
  font-size: var(--step-0);
}

.gif-image{
  width: 200px;
  height: 120px;
}
.gif-image>img{
  display: block;
  width: 100%;
  height: 100%;
}

.gif{
  width: 200px;
  border-radius: 1rem;
  overflow: hidden;
}
.gif-title{
  font-weight: 500;
  font-size: 0.8rem;
  padding: 0.8rem;
  background-color: #eee;
  color: #222;
}