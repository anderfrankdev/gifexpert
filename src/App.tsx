import { useState } from "react";
import addCategory from "./events/addCategory";
import { GifGrid } from "./components/GifGrid";
import "./App.css";

function App() {
  const categoryState = useState([]);
  const [categories] = categoryState;

  return (
    <>
      <h1>GifExpertApp</h1>

      <input
        type="text"
        className="input"
        onKeyUp={addCategory(categoryState)}
        placeholder="Type a topic"
      />

      <button onClick={addCategory(categoryState)} className="button">
        Agregar
      </button>

      <GifGrid categories={categories} />
    </>
  );
}

export default App;
