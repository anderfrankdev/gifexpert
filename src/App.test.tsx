import { describe, expect, it } from "vitest";
import App  from "./App";
import {fireEvent, render, screen} from './utils/test-utils'

describe.concurrent('Tests in <App/>',()=>{

	it('Should match snapshot',async ()=>{

		let component = render(<App />)
		expect(component.container).toMatchSnapshot()

	})

	it('Should render a category',()=>{

		const inputValue = "Goku"

		const {container} = render(<App/>)
		const inputScreen:HTMLInputElement = screen.getByTestId('input')!

		fireEvent.input(inputScreen, {target: {value: inputValue}})
		
		const input:HTMLInputElement = container.querySelector('.input')! 
		
		expect(input.value).toBe(inputValue)

		fireEvent.keyUp(screen.getByTestId('input'),{keyCode:13})

		expect(container.querySelector(".category")).not.toBeNull()
	})
})