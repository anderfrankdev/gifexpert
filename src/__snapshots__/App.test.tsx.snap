// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tests in <App/> > Should match snapshot 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <input
    class="input"
    placeholder="Type a topic"
    type="text"
  />
  <button
    class="button"
  >
    Agregar
  </button>
  <div>
    <p>
      Add a category
    </p>
  </div>
</div>
`;
